src/geventhttpclient/useragent.py

```diff
@@ -314,6 +321,78 @@ def _verify_status(self, status_code, url=None):
         if status_code not in self.valid_response_codes:
             raise BadStatusCode(url, code=status_code)
 
+    def _encode_files(self, files, data):
+        """Build the body for a multipart/form-data request.
+
+        Will successfully encode files when passed as a dict or a list of
+        tuples. Order is retained if data is a list of tuples but arbitrary
+        if parameters are supplied as a dict.
+        The tuples may be 2-tuples (filename, fileobj), 3-tuples (filename, fileobj, contentype)
+        , 4-tuples (filename, fileobj, contentype, custom_headers) or 5-tuples (filename, fileobj, contentype,
+        custom_headers, custom boundary).
+        """
```

Comment:
Is this logic already present somewhere else? (in `requests` or something)

Can we reuse it? If not, can you at least refer to that in the comment (because without any explanation the format sounds very un-obvious :)

Reply:
The format is taken from requests library, I just added 5 tuple version because I saw some people wanting to use custom boundaries

Reply:
I ll add some clarification to comments

Reply:
Done